Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpAppxBuildTools.cs

```diff
@@ -359,15 +389,53 @@ private static string GetMSBuildLoggingCommand(string logDirectory, string logFi
         }
 
         /// <summary>
-        /// Adds the 'Gaze Input' capability to the manifest.
+        /// An overload of AddResearchModeCapability that will read the AppX manifest
+        /// from the build output and update the manifest file with the rescap aka
+        /// research mode capability.
         /// </summary>
-        /// <remarks>
-        /// This is a workaround for versions of Unity which don't have native support
-        /// for the 'Gaze Input' capability in its Player Settings preference location.
-        /// Note that this function is only public to poke a hole for testing - do not
-        /// take a dependency on this function.
-        /// </remarks>
-        public static void AddGazeInputCapability(XElement rootNode)
+        /// <param name="buildInfo">An IBuildInfo containing a valid OutputDirectory</param>
+        public static void AddCapabilities(IBuildInfo buildInfo)
+        {
+            string manifestFilePath = GetManifestFilePath(buildInfo);
+            if (manifestFilePath == null)
+            {
+                throw new FileNotFoundException("Unable to find manifest file");
+            }
+
+            var rootElement = XElement.Load(manifestFilePath);
+            var uwpBuildInfo = buildInfo as UwpBuildInfo;
+
+            Debug.Assert(uwpBuildInfo != null);
+            if (uwpBuildInfo.GazeInputCapabilityEnabled)
+            {
+                AddGazeInputCapability(rootElement);
+            }
+
+            if (EditorUserBuildSettings.wsaSubtarget == WSASubtarget.HoloLens)
+            {
+                if (uwpBuildInfo.ResearchModeCapabilityEnabled)
+                {
+                    AddResearchModeCapability(rootElement);
+                }
+
+                if (EditorUserBuildSettings.wsaGenerateReferenceProjects && uwpBuildInfo.AllowUnsafeCode)
+                {
+                    AllowUnsafeCode(rootElement);
+                }
+            }
+
+            rootElement.Save(manifestFilePath);
+        }
+
+        /// <summary>
+        /// Adds a capability to the given rootNode, which must be the read AppX manifest from
+        /// the build output.
+        /// </summary>
+        /// <param name="rootNode">An <see cref="XElement"/> containing the AppX manifest from 
```

Comment:
So unfortunately our docs validation system doesn't see these crefs as being valid (even though these are fairly well known things).

In order to avoid that, it might be good to just remove the <see cref="XElement" /> part and replace it with it with just XElement.

(Same for XName and XAttribute down below:

[19-09-16 04:37:36.346]Warning:[MetadataCommand.ExtractMetadata]Invalid cref value "!:XElement" found in triple-slash-comments for AddCapability, ignored.
[19-09-16 04:37:36.346]Warning:[MetadataCommand.ExtractMetadata]Invalid cref value "!:XName" found in triple-slash-comments for AddCapability, ignored.
[19-09-16 04:37:36.346]Warning:[MetadataCommand.ExtractMetadata]Invalid cref value "!:XAttribute" found in triple-slash-comments for AddCapability, ignored.)

Reply:
Ok, will do that. Is there a GitHub repo of your docs validation system? So that i can open an issue...i think official Microsoft classes should be recognized by it.

Reply:
For this we just use docfx:

https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/DevDocGuide.html

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_release/docfx.json

I would say it's a bug that we don't currently handle those other types, but in the interest of not blocking behind addressing that (i.e. it could be really easy or a realllllll huge can of worms) it's probably easier in this case just to use the non-cref based text.

Reply:
I got rid of them and will file an issue at dotnet/docfx that points to this pull request. Let's see what happens ^^